src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
+    def __init__(
+            self,
+            ddoc,
+            search_index_name=None,
+            search_index=None
```

Comment:
The only reason not to do this is for consistency across the lib.  The `View` class has it as `view_name`, `map_func` and `reduce_func`.  I suppose that from the "function" perspective we could over look the consistency aspect since in the `View` there are two functions that need naming whereas in the `SearchIndex` there is only one so `func` would be fine there but the naming consistency of the view and the search index still exists.  EIther way this only affects the documentation and most likely no one on the planet would notice other than me.